llm_sandbox/core/mixins.py

```
@@ -44,67 +43,86 @@ def copy_from_container(self, container: Any, src: str) -> tuple[bytes, dict]:
+                if force_kill_on_timeout and callable(handler):
+                    try:
+                        handler()  # pyright: ignore[reportAttributeAccess]
+                    except Exception:  # noqa: BLE001
+                        self.logger.warning("Failed to cleanup container after timeout")
+
+                raise SandboxTimeoutError(msg, timeout_duration=timeout)
 
-        if exception[0]:
-            raise exception[0]
+            if exception[0]:
+                raise exception[0]
```

Pylint (3.3.7)

[error] Raising NoneType while only classes or instances are allowed

(E0702)